src/core.ts

```diff
@@ -61,11 +61,16 @@ export class TemplateResult {
     for (let i = 0; i < l; i++) {
       const s = this.strings[i];
       html += s;
-      // We're in a text position if the previous string closed its tags.
-      // If it doesn't have any tags, then we use the previous text position
-      // state.
-      const closing = findTagClose(s);
-      isTextBinding = closing > -1 ? closing < s.length : isTextBinding;
+      const close = s.lastIndexOf('>');
+      const open = s.indexOf('<', close + 1);
+      // We're in a text position if the previous string closed its last tag, an
+      // attribute position if the string opened an unclosed tag, and unchanged
+      // if the string had no bracets at all:
+      //
+      // "...>...": text position. open === -1, close > -1
+      // "...<...": attribute position. open > -1
+      // "...": no change. open === -1, close === -1
+      isTextBinding = open === -1 && (close > -1 || isTextBinding);
```

Comment:
If we want to get fancy, we can avoid the forward scan in some cases:

```js
const close = s.lastIndexOf('>');
if (close === -1) {
  if (isTextBinding) {
    isTextBinding = s.indexOf('<', close + 1) === -1
  } else {
    // Woop. We were in an attributes, and we didn't close it.
    // No need for forward scan.
    isTextBinding = false;
  }
} else {
  isTextBinding = s.indexOf('<', close + 1) === -1;
}

// Compressed is just a flip of what you already wrote.
const close = s.lastIndexOf('>');
isTextBinding = (close > -1 || isTextBinding) && s.indexOf('<', close + 1) === -1

Reply:
There's a runtime perf / file size balancing game we're playing right now. I'm hoping that the `close` index is generally close enough to the end of the string that both the backward and forward scan is cheap, and we can justify saving a few bytes here.

Reply:
This compresses better than the new code, should save 2 bytes. 😉

```js
// Yours:
var b=a.lastIndexOf(">");c=-1===a.indexOf("<",b+1)&&(-1<b||c)

// Mine:
var b=a.lastIndexOf(">");c=(-1<b||c)&&a.indexOf("<",b+1)
```

It might have gotten lost in the code I posted (edited it to add more new lines), but the compressed (not if-statement) code is at the bottom:

```js
// Compressed is just a flip of what you already wrote.
const close = s.lastIndexOf('>');
isTextBinding = (close > -1 || isTextBinding) && s.indexOf('<', close + 1) === -1
```

That one's faster _and_ smaller.

Reply:
Edit, I forgot the comparison in my compiled output. Updated the comment.